### PR TITLE
[CARBONDATA-2110] Remove tempCsv option in test cases

### PR DESCRIPTION
**Pull request by xuchuanyin**
Since 'TempCSV' option is deprecated in PR #1916 ,

1. the tests using this option should also be updated.
2. log warn message about this deprecated option if user still use it.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
 `NO`
 - [X] Any backward compatibility impacted?
 `NO`
 - [x] Document update required?
`NO`
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
`Deleted the corresponding tests about tempcsv option`
        - How it is tested? Please attach test report.
`Tested it local machine`
        - Is it a performance related change? Please attach the performance test report.
`No`
        - Any additional information to help reviewers in testing this change.
`NO`
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
`Not related`

### CODE REVIEW

**Comment by jackylk** [comment]
LGTM

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDataFrameWriter.scala [hunk 1]
```diff
@@ -42,6 +42,10 @@ class CarbonDataFrameWriter(sqlContext: SQLContext, val dataFrame: DataFrame) {
 
   private def writeToCarbonFile(parameters: Map[String, String] = Map()): Unit = {
     val options = new CarbonOption(parameters)
+    if (options.tempCSV) {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDataFrameWriter.scala]: please add a comment in CarbonOptions.tempCSV to describe it is deprecated. And remove CarbonOptions.compress

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2238/

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDataFrameWriter.scala (hunk 1 above)
Comment (xuchuanyin) [reply integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDataFrameWriter.scala]: OK.
move this deprecate information to `CarbonOptions`

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3479/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2246/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3487/

**Comment by jackylk** [comment]
please rebase

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3344/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2275/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3512/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3359/

**Comment by xuchuanyin** [comment]
@jackylk done

**Comment by jackylk** [comment]
@xuchuanyin master is freeze before 1.3 release. I will merge this after some time

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2828/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4074/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3772/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4144/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2900/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3806/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2981/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4287/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3125/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4359/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5178/

**Pull request closed without merging**